python/tests/test_account.py

```diff
@@ -1215,19 +1232,39 @@ def test_group_many_members_add_leave_remove(self, acfactory, lp):
         ac3 = accounts[1]
         msg3 = ac3.wait_next_incoming_message()
         assert msg3.text == "hello"
-        contacts = ac3.get_contacts()
-        assert len(contacts) == 3
+        ac3_contacts = ac3.get_contacts()
+        assert len(ac3_contacts) == 3
         ac4_contacts = ac3.get_contacts(query=accounts[2].get_config("addr"))
         assert len(ac4_contacts) == 1
 
-        lp.sec("ac1: removing one contacts and checking things are right")
-        to_remove = msg.chat.get_contacts()[-1]
+        lp.sec("ac2: removing one contact")
+        to_remove = contacts[-1]
         msg.chat.remove_contact(to_remove)
 
+        lp.sec("ac1: receiving system message about contact removal")
         sysmsg = ac1.wait_next_incoming_message()
         assert to_remove.addr in sysmsg.text
         assert len(sysmsg.chat.get_contacts()) == 3
 
+        # Receiving message about removed contact does not reset gossip
+        assert chat.get_summary()["gossiped_timestamp"] == gossiped_timestamp
+
+        lp.sec("ac1: sending another message to the chat")
+        chat.send_text("hello2")
+        msg = ac2.wait_next_incoming_message()
+        assert msg.text == "hello2"
+        assert chat.get_summary()["gossiped_timestamp"] == gossiped_timestamp
+
+        lp.sec("ac1: adding fifth member to the chat")
+        chat.add_contact(contact5)
+        # Additng contact to chat resets gossiped_timestamp
+        assert chat.get_summary()["gossiped_timestamp"] >= gossiped_timestamp
```

Comment:
TODO: it is better to test here explicitly that ac5 is able to send encrypted messages to the chat because keys of everyone else have been gossiped in the member addition message

Reply:
Fixed and force-pushed